e2e_test/js-tests/test_viem_smoketest.mjs

```diff
@@ -0,0 +1,170 @@
+import { assert } from "chai";
+import "mocha";
+import {
+	createPublicClient,
+	createWalletClient,
+	http,
+	defineChain,
+} from "viem";
+import { celoAlfajores } from "viem/chains";
+import { privateKeyToAccount } from "viem/accounts";
+import fs from "fs";
+
+// Load compiled contract
+const testContractJSON = JSON.parse(fs.readFileSync(process.env.COMPILED_TEST_CONTRACT, 'utf8'));
+
+
+// Setup up chain
+const devChain = defineChain({
+	...celoAlfajores,
+	id: 1337,
+	name: "local dev chain",
+	network: "dev",
+	rpcUrls: {
+		default: {
+			http: [process.env.ETH_RPC_URL],
+		},
+	},
+});
+
+const chain = (() => {
+	switch (process.env.NETWORK) {
+		case 'alfajores':
+			return celoAlfajores
+		default:
+			return devChain
+	};
+})();
```

Comment:
We could start moving that to a shared file and maybe even include the `publicClient` and `walletClient ` below. It is already used in both `test_viem_tx.mjs` and `send_tx.mjs`.

Reply:
Good point, I was thinking about this.

Reply:
Ok I've made this change.